### PR TITLE
PIMOB-1362(3): Create & Use paymentFormOutcome logging event 

### PR DESCRIPTION
**Pull request by alex-ioja-yang-cko**
## Proposed changes

Create new logging event and trigger on completion of tokenisation request.

Report failure of request via a new `warn`  event.

### CODE REVIEW

File: Source/Core/Logging/FramesLogEvent.swift [hunk 1]
```diff
@@ -56,6 +58,8 @@ enum FramesLogEvent: Equatable, PropertyProviding {
             return "payment_form_presented"
         case .paymentFormSubmitted:
             return "payment_form_submitted"
+        case .paymentFormOutcome:
+            return "payment_form_outcome"
```
Comment (deepesh-vasthimal-cko) [inline Source/Core/Logging/FramesLogEvent.swift:62]: Can we rename this to `payment_form_submitted_result` ?

File: Source/Core/Logging/FramesLogEvent.swift [hunk 2]
```diff
@@ -38,11 +38,13 @@ enum FramesLogEvent: Equatable, PropertyProviding {
     case paymentFormInitialised(environment: Environment)
     case paymentFormPresented
     case paymentFormSubmitted
+    case paymentFormOutcome(token: String)
```
Comment (deepesh-vasthimal-cko) [inline Source/Core/Logging/FramesLogEvent.swift:41]: Can we rename this paymentFormOutcome to paymentFormSubmittedResult ?

File: Source/Core/Logging/FramesLogEvent.swift [hunk 3]
```diff
@@ -96,13 +105,16 @@ enum FramesLogEvent: Equatable, PropertyProviding {
         case let .paymentFormInitialised(environment):
             let environmentString = environment.rawValue == "live" ? "production" : environment.rawValue
             return [.environment: environmentString].mapValues(AnyCodable.init(_:))
+        case let .paymentFormOutcome(token):
+            return [.tokenID: AnyCodable(token)]
```
Comment (deepesh-vasthimal-cko) [inline Source/Core/Logging/FramesLogEvent.swift:109]: 👍

File: Source/UI/PaymentForm/ViewModel/DefaultPaymentViewModel.swift [hunk 4]
```diff
@@ -219,6 +220,15 @@ extension DefaultPaymentViewModel: PaymentViewControllerDelegate {
     guard let viewController = FramesFactory.getBillingFormViewController(style: billingFormStyle, data: billingFormData, delegate: self) else { return }
     sender?.present(viewController, animated: true)
   }
+
+    private func logOutcome(_ result: Result<TokenDetails, TokenisationError.TokenRequest>) {
```
Comment (deepesh-vasthimal-cko) [inline Source/UI/PaymentForm/ViewModel/DefaultPaymentViewModel.swift:224]: may be rename to : `logOutTokenResult`

**Review by deepesh-vasthimal-cko — APPROVED** [review]
Minor renaming suggestions

File: Source/UI/PaymentForm/ViewModel/DefaultPaymentViewModel.swift (hunk 4 above)
Comment (alex-ioja-yang-cko) [reply Source/UI/PaymentForm/ViewModel/DefaultPaymentViewModel.swift]: ✅

File: Source/Core/Logging/FramesLogEvent.swift (hunk 1 above)
Comment (alex-ioja-yang-cko) [reply Source/Core/Logging/FramesLogEvent.swift]: ✅

File: Source/Core/Logging/FramesLogEvent.swift (hunk 2 above)
Comment (alex-ioja-yang-cko) [reply Source/Core/Logging/FramesLogEvent.swift]: ✅

**Review by deepesh-vasthimal-cko — APPROVED** [review]

**Pull request merged**